servers/src/main/java/tachyon/master/MasterInfo.java

```diff
@@ -15,1704 +15,183 @@
    */
-  public InetSocketAddress getMasterAddress() {
-    return mMasterAddress;
+  public List<FileBlockInfo> getFileBlockList(long fileId)
+      throws InvalidPathException, FileDoesNotExistException {
+    return mFileSystemMaster.getFileBlockInfoList(fileId);
   }
 
   /**
-   * Get the MasterSource instance
+   * Gets the dependency info associated with the given id.
    *
-   * @return the MasterSource instance
+   * @param dependencyId id of the dependency
+   * @return the dependency info
+   * @throws DependencyDoesNotExistException when the dependency does not exist
    */
-  public MasterSource getMasterSource() {
-    return mMasterSource;
+  public DependencyInfo getClientDependencyInfo(int dependencyId)
+      throws DependencyDoesNotExistException {
+    return mFileSystemMaster.getClientDependencyInfo(dependencyId);
   }
 
   /**
-   * Get a new user id
+   * Gets the file info for the file at the given path
    *
-   * @return a new user id
+   * @param path path of the file
+   * @return the file info
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
    */
-  public long getNewUserId() {
-    return mUserCounter.incrementAndGet();
+  public FileInfo getFileInfo(TachyonURI path)
+      throws InvalidPathException, FileDoesNotExistException {
+    long fileId = mFileSystemMaster.getFileId(path);
+    return mFileSystemMaster.getFileInfo(fileId);
   }
 
   /**
-   * Get the number of files at a given path.
+   * If the <code>path</code> is a directory, return all the direct entries in it. If the
+   * <code>path</code> is a file, return its FileInfo.
    *
-   * @param path The path to look at
-   * @return The number of files at the path. Returns 1 if the path specifies a file. If it's a
-   *         directory, returns the number of items in the directory.
-   * @throws InvalidPathException when the path is invalid
-   * @throws FileDoesNotExistException when a non-existent file is encountered
+   * @param path the target directory/file path
+   * @return A list of FileInfo
+   * @throws FileDoesNotExistException when the file does not exist
+   * @throws InvalidPathException when an invalid path is encountered
    */
-  public int getNumberOfFiles(TachyonURI path) throws InvalidPathException,
-      FileDoesNotExistException {
-    Inode inode = getInode(path);
-    if (inode == null) {
-      throw new FileDoesNotExistException(path.toString());
-    }
-    if (inode.isFile()) {
-      return 1;
-    }
-    return ((InodeFolder) inode).getNumberOfChildren();
+  public List<FileInfo> getFileInfoList(TachyonURI path)
+      throws FileDoesNotExistException, InvalidPathException {
+    long fileId = mFileSystemMaster.getFileId(path);
+    return mFileSystemMaster.getFileInfoList(fileId);
   }
 
   /**
-   * Get the total number of files.
+   * Gets absolute paths of all in memory files.
```

Comment:
getters can omit the method description if it redundant (applies to other methods in this file)